### PR TITLE
[FIX]base_tier_validation: Selection Field translated

### LINKED ISSUE #822: [16.0+] base_tier_validation: Selection Field not translated.
## base_tier_validation
The name of the module that has a bug.

## Describe the bug
The selection field in the review table is no longer translated. It shows the variable name instead of the text value.

image

It should show:
The `pending` state should show as `Pending`. (Capital P)
And thereby being able to be translated.
Screenshot From V15.
image

## To Reproduce
**Affected versions**: 16, 17

1. Install the base module
2. Install a tier_validation module e.g. sale_tier_validation
3. Create a tier definition
4. Create a document to be validation with a tier validation
5. Look at the tier review table. See that the translation text is not resolved.

**Expected behavior**
The text to be translated as in Pre V16 versions

**Additional context**
As discovered in: https://github.com/OCA/server-ux/pull/793#issuecomment-1883808701

### PR DESCRIPTION
**Pull request by bizzappdev**
Fix: #822

### CODE REVIEW

**Review by bosd — APPROVED** [review]
Code review LGTM

File: base_tier_validation/templates/tier_validation_templates.xml [hunk 1]
```diff
@@ -94,6 +94,7 @@
             <field name="sequence" />
             <field name="requested_by" />
             <field name="status" />
```
Comment (bosd) [inline base_tier_validation/templates/tier_validation_templates.xml:96]: Did not func test it. But maybe only display the field `display_status` here.

Comment (bizzappdev) [reply base_tier_validation/templates/tier_validation_templates.xml:96]: The field will be needed for the conditional class. https://github.com/OCA/server-ux/pull/830/files#diff-8183ea04ccd831a8c08ad90be8130df46e9d358a6b4f330a36f93cd348a51b3cR70

**Review by Dranyel-Bosd — APPROVED** [review]
LGTM :+1:  :sparkler:

**Review by LoisRForgeFlow — APPROVED** [review]
Functional test :+1: 

/ocabot merge minor

**Comment by LoisRForgeFlow** [comment]
@bizzappdev This will need to be fwported to v17, right?

**Comment by bizzappdev** [comment]
@LoisRForgeFlow fix for v17 is also ready https://github.com/OCA/server-ux/pull/839

**Pull request merged**